e2etest/zt_newe2e_file_test.go

```diff
@@ -0,0 +1,247 @@
+		Body: body,
+		ObjectProperties: ObjectProperties{
+			HTTPHeaders: contentHeaders{
+				contentType: pointerTo("text/html"),
+			},
+		},
+	}, false)
+}
+
+func (s *FileTestSuite) Scenario_UploadFileProperties(svm *ScenarioVariationManager) {
+	size := int64(0)
+	fileName := "test_properties"
+	body := NewRandomObjectContentContainer(svm, size)
+
+	srcObj := CreateResource[ContainerResourceManager](svm, GetRootResource(svm, common.ELocation.Local()), ResourceDefinitionContainer{}).
+		GetObject(svm, fileName, common.EEntityType.File()) // awkward capitalization to see if AzCopy catches it.
+	srcObj.Create(svm, body, ObjectProperties{})
+
+	dstObj := CreateResource[ContainerResourceManager](svm, GetRootResource(svm, common.ELocation.File()), ResourceDefinitionContainer{}).
+		GetObject(svm, fileName, common.EEntityType.File())
+
+	metadata := common.Metadata{"Author": pointerTo("gapra"), "Viewport": pointerTo("width"), "Description": pointerTo("test file")}
+	contentType := pointerTo("testctype")
+	contentEncoding := pointerTo("testenc")
+
+	RunAzCopy(svm, AzCopyCommand{
+		Verb:    AzCopyVerbCopy,
+		Targets: []ResourceManager{srcObj, dstObj},
+		Flags: CopyFlags{
+			CopySyncCommonFlags: CopySyncCommonFlags{
+				Recursive: pointerTo(true),
+			},
+			Metadata:        metadata,
+			ContentType:     contentType,
+			ContentEncoding: contentEncoding,
+			NoGuessMimeType: pointerTo(true),
+		},
+	})
+
+	ValidateResource[ObjectResourceManager](svm, dstObj, ResourceDefinitionObject{
+		Body: body,
+		ObjectProperties: ObjectProperties{
+			Metadata: metadata,
+			HTTPHeaders: contentHeaders{
+				contentType:     contentType,
+				contentEncoding: contentEncoding,
+			},
+		},
+	}, false)
+}
+
+func (s *FileTestSuite) Scenario_DownloadPreserveLMTFile(svm *ScenarioVariationManager) {
+	body := NewZeroObjectContentContainer(0)
+	name := "test_upload_preserve_last_mtime"
+	srcObj := CreateResource[ObjectResourceManager](svm, GetRootResource(svm, common.ELocation.File()), ResourceDefinitionObject{ObjectName: pointerTo(name), Body: body})
+	dstObj := CreateResource[ContainerResourceManager](svm, GetRootResource(svm, common.ELocation.Local()), ResourceDefinitionContainer{}).GetObject(svm, name, common.EEntityType.File())
+
+	srcObjLMT := srcObj.GetProperties(svm).FileProperties.LastModifiedTime
+
+	RunAzCopy(
+		svm,
+		AzCopyCommand{
+			Verb:    AzCopyVerbCopy,
+			Targets: []ResourceManager{srcObj, dstObj},
+			Flags: CopyFlags{
+				PreserveLMT: pointerTo(true),
+			},
+		})
+
+	ValidateResource[ObjectResourceManager](svm, dstObj, ResourceDefinitionObject{
+		Body: body,
+		ObjectProperties: ObjectProperties{
+			FileProperties: FileProperties{
+				LastModifiedTime: srcObjLMT,
+			},
+		},
+	}, false)
+}
+
+func (s *FileTestSuite) Scenario_Download63MBFile(svm *ScenarioVariationManager) {
+	body := NewRandomObjectContentContainer(svm, 63*common.MegaByte)
+	name := "test_upload_preserve_last_mtime"
```

Comment:
min: not necessary but it could be good to change name to match the test ("test_63mb_in4mb_upload.txt")